### ISSUE TITLE
"die" conversation endpoint does not work if conversation was started in space, from an "on enter" block

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#3228]
This doesn't work:

```
mission "Shady passenger transport - double cross"
	name "Shady passenger transport"
	description "Discreetly transport <bunks> shady passengers to <destination>."
	minor
	repeat
	source
		attributes frontier "dirt belt" rim north south pirate
	destination
		attributes "near earth" core urban
		distance 5 10
	passengers 6
	on enter
		conversation
			`As soon as you blast off, the cutthroat mercenaries you're transporting try to take control of the ship!`
			choice
				`(Shoot first!)`
					goto shoot
				`(Pay them off.)`
					goto "bribe"
			label shoot
			`Unfortunately, they're faster than you.`
				die
```

Instead of dying, the conversation simply ends.

The conversations wiki page suggests that `die` is an appropriate conversation endpoint and doesn't make any mention of it not working for conversations started while flying around, so this may simply be a bug. Either way, this should work, which would enable missions like the simplified stub above, where your passengers try to wrest control of your ship.

**Comment by tehhowch** [comment]
This is the same issue behind #3206, #3167, and #3129 
There is no callback function that handles the conversation response unless the conversation is an `on offer`

**Comment by tehhowch** [comment]
Possible with #3376

**Comment by endless-sky** [comment]
Addressed by merging #3376.
